relay-server/src/services/processor/event.rs

```diff
@@ -87,8 +89,10 @@ pub fn extract<G: EventProcessing>(
         })
     } else if let Some(item) = transaction_item {
         relay_log::trace!("processing json transaction");
-        state.event_metrics_extracted = item.metrics_extracted();
-        state.spans_extracted = item.spans_extracted();
+        result = Some((
```

Comment:
Here i decided to return an optional, since we are not checking the metrics and spans extracted fields, so instead of forcing people to supply it, I will just return `Some(x)` indicating that the modification of those fields should take place.